test/proxy_funds.js

```diff
@@ -0,0 +1,129 @@
+    factory = await DAOFactory.new(kernelBase.address, aclBase.address, '0x00')
+
+    appCode = await AppStub.new()
+
+    const receipt = await factory.newDAO(permissionsRoot)
+    const kernelAddress = receipt.logs.filter(l => l.event == 'DeployDAO')[0].args.dao
+
+    kernel = Kernel.at(kernelAddress)
+    kernelProxy = getContract('KernelProxy').at(kernelAddress)
+    acl = ACL.at(await kernel.acl())
+
+    const r = await kernel.APP_MANAGER_ROLE()
+    await acl.createPermission(permissionsRoot, kernel.address, r, permissionsRoot)
+
+    // app
+    await kernel.setApp(APP_BASE_NAMESPACE, appId, appCode.address)
+    const initializationPayload = appCode.contract.initialize.getData()
+    appProxy = await AppProxyUpgradeable.new(kernel.address, appId, initializationPayload, { gas: 6e6 })
+    app = AppStub.at(appProxy.address)
+
+    ETH = await appProxy.ETH()
+  })
+
+  const recoverEth = async (proxy, vault) => {
+    const amount = 1
+    const initialBalance = await getBalance(proxy.address)
+    const initialVaultBalance = await getBalance(vault.address)
+    const r = await proxy.sendTransaction({ value: 1, gas: 31000 })
+    assert.equal((await getBalance(proxy.address)).valueOf(), initialBalance.plus(amount))
+    await proxy.transferToVault(ETH)
+    assert.equal((await getBalance(proxy.address)).valueOf(), 0)
+    assert.equal((await getBalance(vault.address)).valueOf(), initialVaultBalance.plus(initialBalance).plus(amount).valueOf())
+  }
+
+  const recoverTokens = async (proxy, vault) => {
+    const amount = 1
+    const token = await getContract('StandardTokenMock').new(accounts[0], 1000)
+    const initialBalance = await token.balanceOf(proxy.address)
+    const initialVaultBalance = await token.balanceOf(vault.address)
+    await token.transfer(proxy.address, amount)
+    assert.equal((await token.balanceOf(proxy.address)).valueOf(), initialBalance.plus(amount))
+    await proxy.transferToVault(token.address)
+    assert.equal((await token.balanceOf(proxy.address)).valueOf(), 0)
+    assert.equal((await token.balanceOf(vault.address)).valueOf(), initialVaultBalance.plus(initialBalance).plus(amount).valueOf())
+  }
+
+  const failWithoutVault = async (proxy, vault) => {
+    const amount = 1
+    const vaultId = hash('vaultfake.aragonpm.test')
+    const initialBalance = await getBalance(proxy.address)
+    await kernel.setDefaultVaultId(vaultId)
+    const r = await proxy.sendTransaction({ value: 1, gas: 31000 })
+    assert.equal((await getBalance(proxy.address)).valueOf(), initialBalance.plus(amount))
+    return assertRevert(async () => {
+      await proxy.transferToVault(ETH)
+    })
+  }
+
+  context('App Proxy', async () => {
+    let vault
+    before(async () => {
+      // vault
+      vault = await getContract('VaultMock').new()
+      const vaultId = hash('vault.aragonpm.test')
+      await kernel.setApp(APP_BASE_NAMESPACE, vaultId, vault.address)
+      await kernel.setDefaultVaultId(vaultId)
```

Comment:
These tests should make and use a proxy for the vault rather than the vault's base contract :).